src/main/java/eladkay/scanner/terrain/TileEntityTerrainScanner.java

```diff
@@ -257,10 +237,9 @@ private boolean checkForBlock(IBlockState local, BlockPos imm) { //True if the b
         if (block.isReplaceable(getWorld(), imm) || block.isAir(local, getWorld(), imm)) { //Replaceable / air
             if (!(block instanceof BlockFluidBase) && !(block instanceof BlockLiquid)) { //Not occupied by liquid
                 return true;
-            }
-            else if (Config.replaceNonSourceLiquid) { //Occupied by liquid, check for config, if config is on then replace non-source blocks
-                if (local.getValue(BlockFluidBase.LEVEL) < 15) //Non-source
-                    return true;
+            } else if (Config.replaceNonSourceLiquid) { //Occupied by liquid, check for config, if config is on then replace non-source blocks
+                //Non-source
+                return local.getValue(BlockFluidBase.LEVEL) < 15;
```

Comment:
Maybe do 
```
return (local.getValue(BlockFluidBase.LEVEL) < 15 || return local.getValue(BlockLiquid.LEVEL) < 15);
```
Just to make sure vanilla liquid works.

Reply:
Fixed in 94527b5.